torch/_inductor/__init__.py

```diff
@@ -25,3 +25,48 @@ def compile(
     from .compile_fx import compile_fx
 
     return compile_fx(gm, example_inputs, config_patches=options)
+
+
+def list_mode_optimizations(mode: str = None) -> Dict[str, Any]:
+    r"""Returns a dictionary describing the optimizations that each of the available
+    modes passed to `torch.compile()` performs.
+
+    Args:
+        mode (str, optional): The mode to return the optimizations for.
+        If None, returns optimizations for all modes
+
+    Example::
+        >>> torch._inductor.list_mode_optimizations()
+    """
+
+    mode_optimizations = {
+        "default": {},
+        "reduce-overhead": {
+            "triton.cudagraphs": False,
+            "size_asserts": False,
+        },
+        "max-autotune": {
+            "epilogue_fusion": False,
+            "max_autotune": False,
+            "triton.cudagraphs": True,
+        },
+    }
+    return mode_optimizations[mode] if mode else mode_optimizations
+
+
+def list_optimizations() -> Dict[str, Any]:
+    r"""Returns a dictionary describing the optimizations and debug configurations
+    that are available to `torch.compile()`.
+
+    The optimizations are documented in `torch._inductor.config`.
+
+    Example::
+
+        >>> torch._inductor.list_optimizations()
+    """
+
+    from torch._inductor import config
+
+    current_config: Dict[str, Any] = config.to_dict()  # type: ignore[attr-defined]
```

Comment:
I am bit worried that some of the configs might not make sense to expose to a user -- though perhaps that is fine.

Reply:
Agreed this is just an easy improvement over telling people go look at config.py

I could create another shortlist of optimizations but that'll duplicate code and be annoying for people to add stuff

or alternatively I could make it clearer in config.py what's a a debugging vs optimization config so something like

```python
class debug
  debug_flag1 = ..
  debug_flag2 = ..

class optimization
  opt_flag1 = ...
```

Reply:
Hrm, for documentation -- a shortlist of optimizations (with text descriptions of what they do and suggestions of what to try) might be easier.

Changing the layout of config.py will be BC breaking, so I'd like to try to avoid it.